joomla/joomla-cms
# Updating checkToken usage to utilise the new JControllerLegacy::checkToken

Pull Request in continuation to #8445 

### Summary of Changes
Implementing the new pattern everywhere. All controllers that extend `JControllerLegacy` are set to use the "nice" handling instead of white screen on a token validation failure. 

Wherever this was within a JSON response, the handling is kept intact as they are better suited it those contexts, changing only the message text.

### Testing Instructions
Test the various backend and front-end forms which would earlier show a blank white page with just a message "Invalid Token". They should now redirect to same page if possible otherwise to home page; with appropriate message.

To see the effect the respective forms should be submitted after manipulating the token input in the form so as to trigger invalid token situation. This can usually be done by selecting in the browser "inspect element" (or similar) and modify the DOM.

If the token is valid (not manipulated or expired), everything should be working as usual without any difference.

The affected components are:
* Banners
* Cache
* Categories
* Contacts
* ContentHistory
* CustomFields
* Finder
* Installer
* Joomla Update
* Language Manager
* Login/Logout
* Media Manager
* Module Manager
* Menu Manager
* Tags
* Template Manager
* User Manager
* Joomla Installation Application (`/installation`)

 
### Documentation Changes Required
Developer documentation should include this feature introduction. Text can be taken from #8445
